contracts/base/Dispatcher.sol

```diff
@@ -33,171 +37,217 @@ abstract contract Dispatcher is Payments, V2SwapRouter, V3SwapRouter, Callbacks,
             } else {
-                if (command == Commands.V2_SWAP_EXACT_IN) {
-                    // equivalent: abi.decode(inputs, (address, uint256, uint256, bytes, bool))
-                    address recipient;
-                    uint256 amountIn;
-                    uint256 amountOutMin;
-                    bool payerIsUser;
+                if (command == Commands.ERC721_PERMIT) {
+                    address spender;
+                    uint256 tokenId;
+                    uint256 deadline;
+                    uint8 v;
+                    bytes32 r;
+                    bytes32 s;
                     assembly {
-                        recipient := calldataload(inputs.offset)
-                        amountIn := calldataload(add(inputs.offset, 0x20))
-                        amountOutMin := calldataload(add(inputs.offset, 0x40))
-                        // 0x60 offset is the path, decoded below
-                        payerIsUser := calldataload(add(inputs.offset, 0x80))
+                        spender := calldataload(inputs.offset)
+                        tokenId := calldataload(add(inputs.offset, 0x20))
+                        deadline := calldataload(add(inputs.offset, 0x40))
+                        v := calldataload(add(inputs.offset, 0x60))
+                        r := calldataload(add(inputs.offset, 0x80))
+                        s := calldataload(add(inputs.offset, 0xA0))
                     }
-                    address[] calldata path = inputs.toAddressArray(3);
-                    address payer = payerIsUser ? lockedBy : address(this);
-                    v2SwapExactInput(map(recipient), amountIn, amountOutMin, path, payer);
-                } else if (command == Commands.V2_SWAP_EXACT_OUT) {
-                    // equivalent: abi.decode(inputs, (address, uint256, uint256, bytes, bool))
-                    address recipient;
-                    uint256 amountOut;
-                    uint256 amountInMax;
-                    bool payerIsUser;
-                    assembly {
-                        recipient := calldataload(inputs.offset)
-                        amountOut := calldataload(add(inputs.offset, 0x20))
-                        amountInMax := calldataload(add(inputs.offset, 0x40))
-                        // 0x60 offset is the path, decoded below
-                        payerIsUser := calldataload(add(inputs.offset, 0x80))
-                    }
-                    address[] calldata path = inputs.toAddressArray(3);
-                    address payer = payerIsUser ? lockedBy : address(this);
-                    v2SwapExactOutput(map(recipient), amountOut, amountInMax, path, payer);
-                } else if (command == Commands.PERMIT2_PERMIT) {
-                    // equivalent: abi.decode(inputs, (IAllowanceTransfer.PermitSingle, bytes))
-                    IAllowanceTransfer.PermitSingle calldata permitSingle;
+
+                    erc721Permit(spender, tokenId, deadline, v, r, s);
```

Comment:
right i wasnt sure! i guess since its probably just going to be used to permit the router for migrating, we can just use address(this)

Reply:
right i wasnt sure! i guess since its probably just going to be used to permit the router for migrating, we can just use address(this)

Reply:
right i wasnt sure! i guess since its probably just going to be used to permit the router for migrating, we can just use address(this)

Reply:
if we remove spender, then tokenId would be the first parameter and it could all be in v3_posm_call
but then we would need to concatenate address(this) to inputs when we call the erc721permit function in v3_posm_call -> leads to more if else statements